api/src/main/java/me/lokka30/treasury/api/economy/response/fail/TreasuryFailureReason.java

```diff
@@ -0,0 +1,343 @@
+package me.lokka30.treasury.api.economy.response.fail;
+
+import me.lokka30.treasury.api.economy.currency.Currency;
+import me.lokka30.treasury.api.economy.misc.EconomyAPIVersion;
+import me.lokka30.treasury.api.economy.response.EconomySubscriber;
+import me.lokka30.treasury.api.economy.response.FailureReason;
+import me.lokka30.treasury.api.economy.transaction.EconomyTransactionInitiator;
+import org.jetbrains.annotations.NotNull;
+
+public enum TreasuryFailureReason implements FailureReason {
```

Comment:
Since this failure reason enum has economy specific entryes and we'll (maybe) have other APIs with their own failure reason this should be named `TreasuryEconomyFailureReason` or `EconomyFailureReason`

Reply:
To add: the FailureReason should also be moved out of the economy package.